Objective-C/CBLReplicator.mm

```diff
@@ -60,11 +60,23 @@ static NSTimeInterval retryDelay(unsigned retryCount) {
     return min(delay, kMaxRetryDelay);
 }
 
+// Replicator progress level types:
 typedef enum {
     kCBLProgressLevelBasic = 0,
     kCBLProgressLevelDocument
 } CBLReplicatorProgressLevel;
 
+// For controlling async start, stop, and suspend:
+typedef enum {
+    kCBLStateStopped = 0,
+    kCBLStateStopping,
+    kCBLStateSuspended,
+    kCBLStateSuspending,
+    kCBLStateUnsuspending,
```

Comment:
may be a single sentence why/what this state

Reply:
Will do in a separated PR. Thinking about enhancing to include offline as the state moving from started -> starting feel weird and we can almost eliminate rawStatus check is most logic.